src/libraries/System.Text.Json/src/System/Text/Json/Serialization/Converters/Collection/DictionaryDefaultConverter.cs

```diff
@@ -39,29 +45,17 @@ protected static JsonConverter<TValue> GetElementConverter(ref ReadStack state)
             return converter;
         }
 
-        protected string GetKeyName(string key, ref WriteStack state, JsonSerializerOptions options)
-        {
-            if (options.DictionaryKeyPolicy != null && !state.Current.IgnoreDictionaryKeyPolicy)
-            {
-                key = options.DictionaryKeyPolicy.ConvertName(key);
-
-                if (key == null)
-                {
-                    ThrowHelper.ThrowInvalidOperationException_NamingPolicyReturnNull(options.DictionaryKeyPolicy);
-                }
-            }
-
-            return key;
-        }
-
         protected static JsonConverter<TValue> GetValueConverter(ref WriteStack state)
         {
-            JsonConverter<TValue> converter = (JsonConverter<TValue>)state.Current.DeclaredJsonPropertyInfo!.ConverterBase;
+            JsonConverter<TValue> converter = (JsonConverter<TValue>)state.Current.JsonClassInfo.ElementClassInfo!.PropertyInfoForClassInfo.ConverterBase;
             Debug.Assert(converter != null); // It should not be possible to have a null converter at this point.
 
             return converter;
         }
 
+        protected JsonConverter<TKey> GetKeyConverter(JsonSerializerOptions options)
```

Comment:
I tried the same approach on `GetValueConverter` but it seems that switching from an static method to an instance one decreases performance, hence I had to avoid caching for the `JsonConverter<TValue>`.

Reply:
Instead of moving this to a instance method, I kept this static method and cached the converter in the call site and that does not regress performance. Thanks for the suggestion.